### PR TITLE
fix(ci): count only changed lines in coverage gate; exclude deletions from jscpd

### LINKED ISSUE #427: CI: coverage gate counts entire file instead of changed lines; duplication gate crashes on deletions
Two related defects in `.github/workflows/ci.yml`:

**Coverage gate (`New code coverage gate` step)** counts every instrumented (`DA:`) line in every changed file rather than only the lines this PR actually added or modified. The effect: a PR that adds 19 well-tested lines to a 500-line file with legacy 60% coverage trips the 80% gate, even when the new code is 100% covered. This contradicts MEMORY.md note from v1.1.8 ("measures only new/changed lines, skips <10 lines"); artifact-keeper-web was never updated.

**Duplication gate (`Code duplication gate` step)** passes the full `git diff --name-only` output to jscpd, which includes deleted file paths. jscpd then crashes with `ENOENT: no such file or directory, lstat ...` because it tries to follow the deleted path. Example failure: PR #414 deletes `src/lib/constants/webhook.ts` and the gate dies before producing a report.

**Fix**:
- Coverage gate: parse `git diff --unified=0` hunks to build a {file → added-line-set} map, intersect with lcov `DA:` line numbers, and gate on hit/(hit+miss) restricted to those lines. Also adopt the <10 changed-lines skip already used in artifact-keeper.
- Duplication gate: add `--diff-filter=ACMR` to `git diff --name-only` so deleted paths never reach jscpd.

Both defects currently block legitimate PRs (notably #414) from merging.

### PR DESCRIPTION
**Pull request by brandonrc**
## Summary

Two related defects in `.github/workflows/ci.yml`:

1. **Coverage gate** counted every instrumented `DA:` line in every changed file, not the lines this PR actually changed. A PR that added 19 well-tested lines to a 500-line file with legacy 60% coverage tripped the 80% threshold even when the new code was fully covered. This contradicts the v1.1.8 intent recorded in the workspace memory ("measures only new/changed lines, skips <10 lines"). artifact-keeper-web's gate was never updated to match.

2. **Duplication gate** passed the raw `git diff --name-only` output to jscpd, which includes deleted file paths. jscpd then crashed with `ENOENT: no such file or directory, lstat ...` because it tried to lstat a path that no longer exists. PR #414 hit this because it deletes `src/lib/constants/webhook.ts`.

## Fix

- **Coverage gate**: parse `git diff --unified=0` hunks per changed file to build a `{file -> added-line-set}` map, then only count lcov `DA:` entries whose line numbers fall in that set. Adopt the `<10 changed lines` skip already in use in the backend repo.
- **Duplication gate**: add `--diff-filter=ACMR` to `git diff --name-only` so deleted paths are excluded before being passed to jscpd.

Both gates still fail the build when they detect a real regression; they just stop spuriously failing PRs that touch legacy code or delete files.

## Test Checklist
- [x] Manually validated the hunk parser against a sample unified diff
- [ ] Unit tests added/updated (N/A: workflow YAML, no unit-test harness)
- [ ] E2E Playwright tests added/updated (N/A)
- [x] No regressions in existing tests (this PR only changes ci.yml)

## UI Changes
- [x] N/A - no UI changes

Closes #427

### CODE REVIEW

**Comment by brandonrc** [comment]
## Multi-agent review summary (3/3 reviewers)

**Reviewer 1 (Code Reviewer)**: APPROVE - hunk parser correctly handles unified diff edge cases (@@ +1 @@ no-comma, @@ -0,0 +1,5 @@ for new files, multi-hunk files); --diff-filter=ACMR correctly excludes deleted (D), typechange (T), unmerged (U), unknown (X), broken (B) - covers all real cases. Shell injection safer than before (env var vs string interpolation).

**Reviewer 2 (audit-devops)**: APPROVE - YAML parses correctly. Python heredoc uses 'PY' delimiter inside bash multiline; subprocess.run inherits cwd correctly. export CHANGED_FILES is correctly scoped. python3 default on ubuntu-latest. No impact on existing Coverage summary step.

**Reviewer 3 (Evidence Collector)**: APPROVE - regression test confirmed: 50 untested new lines still trips gate. Edge cases handled: pure addition, single-line, multi-hunk, pure deletion. Rename-with-content-change verified empirically. Three hypothetical PRs traced through actual parser+lcov logic, all behave correctly.

Verdict: ready to merge when CI green. Unblocks #414.
